src/RevisionPopupWidget.js

```diff
@@ -39,49 +44,83 @@ function getSizeHtml( size ) {
 		'</span>';
 }
 
+/**
+ * Get markup for the edit summary.
+ * @param {Object} data As returned by Api.prototype.getTokenInfo().
+ * @return {string}
+ */
+function getCommentHtml( data ) {
+	if ( data.comment === '' ) {
+		// No edit summary.
+		return '';
+	} else if ( data.comment === undefined ) {
+		// Not yet available.
+		return `
+			<div class="ext-wwt-revisionPopupWidget-comment">
+				<div class="ext-wwt-shimmer ext-www-shimmer-animation"></div>
+				<div class="ext-wwt-shimmer ext-www-shimmer-animation"></div>
+			</div>`;
+	}
+
+	return `
+		<div class="ext-wwt-revisionPopupWidget-comment ext-wwt-revisionPopupWidget-comment-transparent">
+			<span class="comment comment--without-parentheses ext-wwt-revisionPopupWidget-comment">${Tools.bidiIsolate( data.comment, true )}</span>
+			${getSizeHtml( data.size )}
+		</div>`;
+}
+
+/**
+ * Get jQuery objects for the user links.
+ * @param {Object} data As returned by Api.prototype.getTokenInfo().
+ * @return {jQuery}
+ */
+function getUserLinksHtml( data ) {
+	const contribsUrl = mw.util.getUrl( `Special:Contributions/${data.username}` ),
+		// We typically link to Special:Contribs for IPs.
+		userPageUrl = data.isIP ? contribsUrl : mw.util.getUrl( `User:${data.username}` );
+
+	return $( [] )
+		.add(
+			$( '<a>' )
+				.attr( 'href', userPageUrl )
+				.append( Tools.bidiIsolate( data.username ) )
```

Comment:
I wrote a lengthier comment about this below, but I just want to verify; this data.username will always be rendered as HTML even if it's purposefully not supposed to. Is that okay? I think we have most relevant characters illegal in MW to prevent issues with that, but I'm actually not sure.

Reply:
I don't think it's possible to have a username that is interpreted by the browser as anything other than plaintext. We should be safe :) But we can escape this too, if you want.

Reply:
Nah, I wasn't sure, and while I remembered that we prevent the characters that involve illegal titles, I couldn't find anything on mw.org that explained whether we actually allow for something like &nbsp; or something. Thanks for confirming!